game/yodesk.ksy

```diff
@@ -1,564 +1,923 @@
-        35: games_won_is_greater_than
-  instruction:
-    seq:
-      - id: opcode
-        type: u2
-        enum: instruction_opcode
-      - id: arguments
-        type: s2
-        repeat: expr
-        repeat-expr: 5
-      - id: len_text
-        type: u2
-      - id: text
-        type: str
-        size: len_text
-    enums:
-      instruction_opcode:
-        0: place_tile
-        1: remove_tile
-        2: move_tile
-        3: draw_tile
-        4: speak_hero
-        5: speak_npc
-        6: set_tile_needs_display
-        7: set_rect_needs_display
-        8: wait
-        9: redraw
-        10: play_sound
-        11: stop_sound
-        12: roll_dice
-        13: set_counter
-        14: add_to_counter
-        15: set_variable
-        16: hide_hero
-        17: show_hero
-        18: move_hero_to
-        19: move_hero_by
-        20: disable_action
-        21: enable_hotspot
-        22: disable_hotspot
-        23: enable_monster
-        24: disable_monster
-        25: enable_all_monsters
-        26: disable_all_monsters
-        27: drop_item
-        28: add_item
-        29: remove_item
-        30: mark_as_solved
-        31: win_game
-        32: lose_game
-        33: change_zone
-        34: set_shared_counter
-        35: add_to_shared_counter
-        36: set_random
-        37: add_health
-  monster:
-    seq:
-      - id: character
-        type: u2
+          size: _parent.size
+    sounds:
+      doc: |
+        This section declares sounds used in the game. The actual audio data is stored in wav files on the disk (in a directory named `sfx`) so this section contains paths to each sound file.
+        Sounds can be referenced from the scripting language (see `play_sound` instruction opcode below) and from weapon (see `character` structure).
+        Some sound ids (like the one when the hero is hit, or can't leave a
+        zone) are hard coded in the game.
+      seq:
+        - id: count
+          type: s2
+        - id: sounds
+          type: prefixed_strz
+          repeat: expr
+          repeat-expr: -count
+    tile_names:
+      seq:
+        - id: names
+          doc: |
+            List of tile ids and their corresponding names. These are shown in the inventory or used in dialogs (see `speak_hero` and `speak_npc` opcodes).
+          type: tile_name
+          repeat: until
+          repeat-until:  _.tile_id == 0xFF_FF
+    tile_name:
+      seq:
+        - id: tile_id
+          type: u2
+        - id: name
+          type: strz
+          size: 0x18
+          if: tile_id != 0xFF_FF
+    tiles:
+      seq:
+        - id: tiles
+          type: tiles_entries
+          size: _parent.size
+    tiles_entries:
+      seq:
+        - id: tiles
+          type: tile
+          repeat: eos
+    tile:
+        seq:
+          - id: attributes
+            doc: |
+              Bit mask of various tile attributes. Most bits are purely informational and do not have any effect on game play.
```

Comment:
Why not to specify it in KS using bit-sized types instead of putting that info into `doc`?

Reply:
I agree with @KOLANICH. This is a part of the format structure and should be described by KSY. 

@cyco The best way to do that is to use little-endian bit-sized integers (https://doc.kaitai.io/user_guide.html#bit-ints-le). In short, it's about adding this into the top-level `meta` section

```diff
   ks-version: 0.9
+  bit-endian: le
   endian: le
   encoding: ASCII
```

and then you just start writing the bit-sized members in the order from the least significant bit in the bitfield ("bit 0"):

```ksy
types:
  flags:
    seq:
      - id: is_opaque # Bit 0
        type: b1
      - id: is_floor # Bit 1
        type: b1
      - id: is_object # Bit 2
        type: b1
      # etc. 
```

Reply:
Hey, I've updated the PR. Tile attributes are now encoded using bit sized integers as you suggested. Please take a look and let me know what you think.

My editor somehow messed up the indentation of the file, I hope the large diff is not a problem.